src/main/java/com/example/java/test/junior/developer/mapper/UserMapper.java

```diff
@@ -0,0 +1,56 @@
+package com.example.java.test.junior.developer.mapper;
+
+import com.example.java.test.junior.developer.dto.UserDto;
+import com.example.java.test.junior.developer.dto.UserRequestDto;
+import com.example.java.test.junior.developer.model.User;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+
+@Component
+public class UserMapper {
+    public User toModel(UserDto userDto) {
+        return User.builder()
+                .firstName(userDto.getFirstName())
+                .lastName(userDto.getLastName())
+                .email(userDto.getEmail())
+                .build();
+    }
+
+    public UserDto toDto(User user) {
+        return UserDto.builder()
+                .id(user.getId())
+                .firstName(user.getFirstName())
+                .lastName(user.getLastName())
+                .email(user.getEmail())
+                .build();
+    }
+
+    public UserDto toDto(UserRequestDto userRequestDto) {
+        return UserDto.builder()
+                .id(userRequestDto.getId())    // Чи потрібно вносити id з запиту ?
+                .firstName(userRequestDto.getFirstName())
+                .lastName(userRequestDto.getLastName())
+                .email(userRequestDto.getEmail())
+                .build();
+    }
+
+    public UserRepresentation toUserRepresentation(UserRequestDto userRequestDto) {
+        // Чи потрібно повертати звідси UserRepresentation ?
+        var user = new UserRepresentation();
+        user.setId(user.getId());
```

Comment:
You don't need to set id here. Id will be autogenerate by keycloak. Remove this line.